rfcs/0000-rfc-forms.md

```diff
@@ -0,0 +1,148 @@
+---
+Status: Active
+Champions: christian24
+PR: {{ update_with_pr_number }}
+---
+
+# @lit-labs/forms
+
+Add advanced form validation for web components. This is a followup to the discussion in the lit repository: https://github.com/lit/lit/discussions/2489 
+
+## Objective
+
+Provide an easy to use dev experience for validating forms with web components in them. 
+
+### Goals
+- Provide a common API, so developers can quickly start using the @lit-labs/forms package without changing their existing controls. 
+- Allow cross field validation
+- Allow async validation
+- get and set dirty, touched, valid states for individual controls
+- allow for configuration (e. g. change on `blur` or `change`)
+- It should be typed
+- It should be possible to set initial values for individual controls. 
+- subscribe to value changes and update dependant controls accordinly. 
+- validate all controls of a form at the same time. 
+- Allow for dynamic forms (e. g. arrays of controls etc.)
+
+### Non-Goals
+- The goal is not to provide a set of Validators 
```

Comment:
Tried clarifying this as well. A validator is just a pure function. They can be shared, but my goal would not be to create a package of validators like email, telephone etc. This can be done in user land by another package.

Reply:
@Christian24 I wouldn't say "pure" because it doesn't have to be a pure function. An async validator is almost never a pure function